### PR TITLE
Update documentation for OpenSSL 3.1 and update to 3.1.1

### PR DESCRIPTION
**Pull request by dcarpente**
This PR has a couple of changes:
1. Update OpenSSL 3.1 submodule to point to 3.1.1 release. https://github.com/openssl/openssl/releases/tag/openssl-3.1.1
2. Update build flags in update.make to disable any new features added with OpenSSL 3.1 we don't need (e.g. SIV block cipher)
3. Update documentation to reflect differences in OpenSSL 3.1.

This is my last planned change for adding support for OpenSSL 3.1. So if there's any missing changes reviewers identify, please leave a comment! Thank you.

### CODE REVIEW

**Comment by dcarpente** [comment]
bors try

File: docs/OpenSSLSupport.md [hunk 1]
```diff
@@ -86,9 +87,28 @@ In addition, OpenSSL by default disables the following algorithms/features
...
+- no-ssl-trace
+  - Disable SSL Trace capabilities. (OpenSSL 1.1.1 builds without SSL Trace by default.)
+- no-ssl
+  - `no-ssl` is synonymous with `no-ssl3`; `no-ssl2` is deprecated.
+- no-uplink
+  - Disable UPLINK interface.
+
+# OpenSSL 3.1 Support
+
+The default OpenSSL enclave build option uses version 1.1.1 of OpenSSL. There are seperate libraries and headers for building OpenSSL 3.1 in an enclave.
```
Comment (achamayou) [inline docs/OpenSSLSupport.md:111]:
```suggestion
The default OpenSSL enclave build option uses version 1.1.1 of OpenSSL. There are separate libraries and headers for building OpenSSL 3.1 in an enclave.
```

File: tests/openssl/tests_3.unsupported [hunk 2]
```diff
@@ -53,12 +53,16 @@
 ../../3rdparty/openssl/openssl_3/test/moduleloadtest.c
 # OSSL_METHOD_STORE methods can call rdtsc directly
 ../../3rdparty/openssl/openssl_3/test/property_test.c
-# Tests have memory leak. https://github.com/openssl/openssl/issues/20716
+# CMP is disabled by OE.
+# Also, some of these have memory leak. https://github.com/openssl/openssl/issues/20716
```
Comment (achamayou) [inline tests/openssl/tests_3.unsupported:57]: The leaks seem to have been fixed now.

**Review by achamayou — APPROVED** [review]

File: tests/openssl/tests_3.unsupported (hunk 2 above)
Comment (dcarpente) [reply tests/openssl/tests_3.unsupported:57]: Thank you for the catch!

**Review by mingweishih — APPROVED** [review]

**Comment by dcarpente** [comment]
bors try

**Comment by radhikaj** [comment]
Thanks for reviewing @achamayou and @mingweishih

**Comment by dcarpente** [comment]
bors try

**Comment by radhikaj** [comment]
Bors r+

**Pull request merged**